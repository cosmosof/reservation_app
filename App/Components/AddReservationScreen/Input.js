// @flow

import * as React from 'react';
import {Text, TextInput, View} from 'react-native';

import styles from './Styles/InputStyles';

type Props = {
  error?: boolean,
  inputStyles?: Object,
  label?: string,
  name?: string,
  onChange: Function,
  onTouch: Function,
  styles: Object,
  value?: string,
};

type State = {
  boxShadow: Object | null,
};

export default class Input extends React.Component<Props, State> {
  static defaultProps: Object;
  constructor(props: Object) {
    super(props);
    this.state = {
      boxShadow: null,
    };
  }
  handleChange = (value: string) => {
    this.props.onChange(this.props.name, value);
  };

  handleTouch = () => {
    this.setState({boxShadow: null});
    this.props.onTouch(this.props.name);
  };
  textInputonFocusStyleHandle() {
    this.setState({boxShadow: styles.boxShadow});
  }
  render() {
    const {boxShadow} = this.state;
    const {error, inputStyles, label, value} = this.props;
    const {container, errorText, textInput} = styles;
    return (
      <View style={container}>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          maxLength={600}
          testID="TextInput"
          onBlur={this.handleTouch}
          onChangeText={this.handleChange}
          onFocus={() => this.textInputonFocusStyleHandle()}
          placeholder={label}
          returnKeyType="next"
          style={[textInput, inputStyles, boxShadow]}
          underlineColorAndroid="transparent"
          value={value}
        />
        {error && <Text style={errorText}>{error}</Text>}
      </View>
    );
  }
}

Input.defaultProps = {
  error: false,
  inputStyles: undefined,
  label: undefined,
  name: undefined,
  value: undefined,
};
