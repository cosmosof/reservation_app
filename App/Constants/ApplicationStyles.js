import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
