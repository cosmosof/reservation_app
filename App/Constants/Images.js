const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
