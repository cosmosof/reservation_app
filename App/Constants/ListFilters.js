export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
