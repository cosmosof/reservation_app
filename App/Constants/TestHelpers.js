const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
