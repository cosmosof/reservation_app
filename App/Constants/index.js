import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
