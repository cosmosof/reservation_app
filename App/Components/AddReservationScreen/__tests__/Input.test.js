import React from 'react';
import {shallow} from 'enzyme';
import Input from '../Input';
import {TestHelpers} from '../../../Constants';

const {noop, sampleText} = TestHelpers;

describe('<Input>', () => {
  const Component = (<Input
    onChange={noop}
    onTouch={noop}
  />);
  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if label prop is set', () => {
      const Clone = React.cloneElement(Component, {label: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper.instance().props.label).toEqual(sampleText);
    });
    it('renders correctly if name prop is set', () => {
      const Clone = React.cloneElement(Component, {name: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper.instance().props.name).toEqual(sampleText);
    });
    it('renders correctly if inputStyles prop is set', () => {
      const Clone = React.cloneElement(Component, {inputStyles: {backgroundColor: 'blue'}});
      const wrapper = shallow(Clone);
      expect(wrapper.instance().props.inputStyles).toEqual({backgroundColor: 'blue'});
    });
    it('renders correctly if value prop is set', () => {
      const Clone = React.cloneElement(Component, {value: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper.instance().props.value).toEqual(sampleText);
    });
  });
  describe('Functionality', () => {
    it('sets state correctly', () => {
      const wrapper = shallow(Component);
      wrapper.setState({boxShadow: {borderColor: 'red'}});
      expect(wrapper.state('boxShadow')).toEqual({borderColor: 'red'});
    });
  });
});
