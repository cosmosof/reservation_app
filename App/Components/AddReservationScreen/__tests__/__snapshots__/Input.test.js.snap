// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Structure renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    keyboardType="default"
    maxLength={600}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    returnKeyType="next"
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#CCCCCC",
          "borderRadius": 2,
          "borderWidth": 0.5,
          "color": "#2d2d2d",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "height": 36,
          "padding": 10,
          "width": 200,
        },
        undefined,
        null,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
